src/main/java/com/unbxd/SchemaManager/Dao/Dao.java

```diff
@@ -0,0 +1,18 @@
+package com.unbxd.SchemaManager.Dao;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.unbxd.SchemaManager.Models.Field;
+import com.unbxd.SchemaManager.Models.SiteSchema;
+
+import java.io.IOException;
+
+public interface Dao {
+
+    public void addNewSchema(SiteSchema schema) throws JsonProcessingException;
+    public SiteSchema getSchemaForSite(String siteKey) throws IOException;
```

Comment:
method should not throw jsonprocessing exeption as it is specific to jackson library
instead create a schemaexception which will be a wrapper to all underlying exceptions